src/clientfactory.ts

```diff
@@ -64,30 +62,28 @@ export class DiscordClientFactory {
             messageCacheLifetime: 5,
             sync: false,
         });
-        return new Bluebird<string>((resolve, reject) => {
-            client.on("ready", async () => {
-                const id = client.user.id;
-                await client.destroy();
-                resolve(id);
-            });
-            client.login(token).catch(reject);
-        }).timeout(READY_TIMEOUT).catch((err: Error) => {
-            log.warn("Could not login as a normal user.", err.message);
-            throw Error("Could not retrieve ID");
+
+        await client.login(token);
+        const id = client.user.id;
+        client.destroy().catch(() => {
```

Comment:
Why not use await / try...catch?

```suggestion
        try {
            await client.destroy();
        } catch (err) {
            log.warn("Failed to destroy client ", id);
        }
```